SparseGrids/tsgLinearSolvers.cpp

```diff
@@ -167,6 +167,68 @@ void TasmanianTridiagonalSolver::decompose(int n, double d[], double e[], double
     }
 }
 
+void TasmanianFourierTransform::dft(const int rank, const int n[], const std::complex<double> data[], std::complex<double> out[], const int direction){
+    // Fourier transform code by ZBM
+    // DOES NOT NORMALIZE the outgoing Fourier coefficients
+
+    if (direction != 1 && direction != -1) std::cerr << "ERROR: direction must be -1 (forward) or 1 (backward); value is " << direction << std::endl;
+    int num_data = 1;
+    for(int r=0; r<rank; r++){
+        num_data *= n[r];
+    }
+
+    std::complex<double> i(0.0, 1.0);
```

Comment:
If the complex constant is defined in `std::complex::I`, then there is no need to redefine it here, just use it. I think it will work if you just say:
```
using std::complex::I;
```
and then use `I` afterwards.

Reply:
I get an error with `std::complex::I`. I looked on Stackexchange but couldn't find `sqrt(-1)` hardcoded somewhere as a constant within `std::complex` in C++11.

Reply:
@zbmorrow I thought we had a problem with `I` being declared as `sqrt(-1)` and that's why you had to remove the `I` that was used for integral. Then the problem went away when we stopped using `using namespace std::complex`

Reply:
That was from `<complex.h>`, which defined `I` as a macro (http://pubs.opengroup.org/onlinepubs/009695399/basedefs/complex.h.html). The problem stopped when we changed `<complex.h>` to `<complex>`.